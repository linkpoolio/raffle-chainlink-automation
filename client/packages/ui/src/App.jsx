import React from 'react'
import { Switch, Route, Redirect } from 'react-router-dom'

import { RaffleList } from '@ui/features/raffle'

export function App() {
  return (
    <>
      <Switch>
        <Route exact path="/">
          <RaffleList />
        </Route>
        <Redirect to="/" />
      </Switch>
    </>
  )
}
