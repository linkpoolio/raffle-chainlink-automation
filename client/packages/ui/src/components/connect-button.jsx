import React from 'react'
import {
  Box,
  Button,
  Modal,
  ModalBody,
  ModalCloseButton,
  ModalContent,
  ModalFooter,
  ModalHeader,
  ModalOverlay,
  Stack,
  useDisclosure
} from '@chakra-ui/react'
import { goerli, useConnect } from 'wagmi'

export const ConnectButton = () => {
  const { connect, error, connectors, isLoading, pendingConnector } =
    useConnect({
      chainId: goerli.id
    })
  const { isOpen, onOpen, onClose } = useDisclosure()

  return (
    <>
      <Button onClick={onOpen} variant="base"></Button>
      <Modal isOpen={isOpen} onClose={onClose}>
        <ModalOverlay />
        <ModalContent bg="rgb(55,91,210)">
          <ModalHeader color="white">Connect</ModalHeader>
          <ModalCloseButton />
          <ModalBody>
            <Stack>
              {connectors &&
                connectors.map((connector) => (
                  <Button
                    variant="wallet"
                    disabled={!connector.ready}
                    key={connector.id}
                    onClick={() => connect({ connector })}>
                    {connector.name}
                    {isLoading &&
                      pendingConnector?.id === connector.id &&
                      ' (connecting)'}
                  </Button>
                ))}
              {error && <Box>{error?.message ?? 'Failed to connect'}</Box>}
            </Stack>
          </ModalBody>
          <ModalFooter />
        </ModalContent>
      </Modal>
    </>
  )
}
