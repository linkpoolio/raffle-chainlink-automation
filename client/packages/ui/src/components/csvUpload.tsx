import React from 'react'

// TODO: build this component
// TODO: build transformer to flat array
// TODO: build transformer to bytes32 each array item
export const CSVUpload = () => {
  return <h3>CSV Upload</h3>
}
