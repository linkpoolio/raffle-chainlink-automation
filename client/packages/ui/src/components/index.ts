export * from './csvUpload'
export * from './error'
export * from './hero'
export * from './loading'
export * from './loadingList'
export * from './modal'
export * from './success'
