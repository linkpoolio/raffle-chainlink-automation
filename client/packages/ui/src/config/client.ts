import { createClient } from 'wagmi'
import { getDefaultClient } from 'connectkit'
import { mainnet, goerli, polygon, optimism, arbitrum } from 'wagmi/chains'

export const client = createClient(
  getDefaultClient({
    appName: 'Raffle Manager',
    chains: [mainnet, goerli, polygon, optimism, arbitrum]
  })
)
