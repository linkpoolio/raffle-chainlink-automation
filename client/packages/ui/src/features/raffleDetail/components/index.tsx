export * from './checkStatusButton'
export * from './detail'
export * from './join'
export * from './joinButton'
export * from './participantStatus'
export * from './provideIdentifier'
export * from './stepManager'
