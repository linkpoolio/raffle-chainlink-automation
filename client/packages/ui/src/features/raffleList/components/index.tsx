export * from './list'
export * from './filters'
export * from './row'
