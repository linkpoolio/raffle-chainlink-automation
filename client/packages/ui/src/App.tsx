import React from 'react'
import { Switch, Route, Redirect } from 'react-router-dom'

import { Routes } from '@ui/Routes'
import { RaffleList } from '@ui/features/raffleList'
import { RaffleDetail } from '@ui/features/raffleDetail'
import { RaffleCreate } from '@ui/features/raffleCreate'
import { AuthenticatedRoute, Hero } from '@ui/components'

export const App = () => (
  <>
    <Switch>
      <Route
        exact
        path={Routes.RaffleList}
        render={(props) => (
          <>
            <Hero />
            <RaffleList {...props} />
          </>
        )}
      />

      <Route
        path={Routes.RaffleDetail}
        render={({ match }) => <RaffleDetail id={match.params.id} />}
      />

      <Route exact path={Routes.RaffleCreate}>
        <AuthenticatedRoute connected={true}>
          <RaffleCreate />
        </AuthenticatedRoute>
      </Route>

      <Redirect to={Routes.RaffleList} />
    </Switch>
  </>
)
