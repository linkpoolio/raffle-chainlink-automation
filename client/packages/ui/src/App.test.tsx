import React from 'react'
import { createRoot } from 'react-dom/client'
import { act } from '@testing-library/react'
import renderer from 'react-test-renderer'

import { App } from '@ui/App'
import { Providers } from '@ui/providers'
import { NavigationBar } from '@ui/components'

const getComponent = () => {
  const component = (
    <Providers>
      <NavigationBar />
      <App />
    </Providers>
  )

  return component
}

describe('App', () => {
  it('renders the App.', () => {
    const container = document.createElement('div')
    document.body.appendChild(container)

    const root = createRoot(container)

    act(() => root.render(getComponent()))

    expect(document.body.textContent).toBeTruthy()
  })

  it('matches snapshot', () => {
    const tree = renderer.create(getComponent()).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
