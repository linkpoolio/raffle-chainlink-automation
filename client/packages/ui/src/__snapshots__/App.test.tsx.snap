// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
[
  <h2>
    RaffleList
  </h2>,
  <a
    href="/raffle/1"
    onClick={[Function]}
  >
    Raffle Detail
  </a>,
]
`;
